snappy/binaries.go

```diff
@@ -154,7 +140,7 @@ func addPackageBinaries(s *snap.Info) error {
 			return err
 		}
 
-		if err := osutil.AtomicWriteFile(generateBinaryName(app), []byte(content), 0755, 0); err != nil {
+		if err := osutil.AtomicWriteFile(app.WrapperPath(), []byte(content), 0755, 0); err != nil {
```

Comment:
This is something to do post release but service files could be managed with EnsureDirState. It's such a nice way to just describe what should be present and not worry about details (one tested implementation). The same applies to app wrappers.